### PR TITLE
Fix andAnyOthers() to properly match earlier expectations

### PR DESCRIPTION
**Pull request by danielsd**
This PR fixes the behavior in https://github.com/mockery/mockery/issues/1051

Additionally I updated an erroneous test case and added a test case for another edge case.

### CODE REVIEW

File: tests/Mockery/ExpectationTest.php [hunk 1]
```diff
@@ -428,15 +428,15 @@ public function testExpectsStringArgumentExceptionMessageDifferentiatesBetweenNu
     {
         $this->mock->shouldReceive('foo')->withArgs(array('a string'));
         $this->expectException(\Mockery\Exception::class);
-        $this->expectExceptionMessageRegExp('/foo\(NULL\)/');
+        $this->expectExceptionMessageMatches('/foo\(NULL\)/');
```
Comment (danielsd) [inline tests/Mockery/ExpectationTest.php:431]: PHPUnit was failing these tests because this method is deprecated.

**Comment by danielsd** [comment]
I was going to add notes to the changelog but noticed there is no log for `1.3.2` yet. Is that intentional? Is there another place to add changelog notes?

**Comment by davedevelopment** [comment]
Hi @danielsd, thank you for this. People usual forget about the CHANGELOG, so I usually put it together before release. If you would like to include a note, feel free to add a `1.3.2 (XXXX-XX-XX)` heading :+1:

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,7 +27,7 @@
 
 ## 1.2.2 (2019-02-13)
 
-* Fix a BC breaking change for PHP 5.6/PHPUnit 5.7.27 (#947)
```
Comment (danielsd) [inline CHANGELOG.md:27]: My editor automatically trims trailing whitespace and I figured it wouldn't hurt to commit these changes.

**Comment by danielsd** [comment]
@davedevelopment Thanks for the info. I went ahead and started the `1.3.2` changelog. Let me know if anything else is needed to make this PR merge-able.

**Comment by davedevelopment** [comment]
Perfect, thank you @danielsd
